docs/specs/apb.yml-plan-support.md

```diff
@@ -0,0 +1,165 @@
+In this case, PlanID is passed to APBs as `_apb_plan_id`. APBs should be written
+expecting this parameter and branch behavior based on its value.
+* Enforce validations at the registry level. If a validation fails, warn spec
+is not conformant and filter from available set. Validations:
+```
+-> len(spec.Plans) > 0
+-> plan.name must be unique amongst all plans
+```
+
+## APB changes
+
+1) New `plans` section of type `[]Plan`. **At least one plan is **required**
+2) Move `parameters` from top level onto `Plan`s, type. New types:
+
+**Plan**
+
+Field Name | Type | Required | Default | Description
+---|---|---|---|---
+name | string| yes |  | Name of the plan, used as an identifier. **Must be unique amongt plans on the apb**
+description | string | yes | | A human readable description of the plan.
+free | bool | yes | | Indicates whether the plan has an associated cost
+metadata | `PlanMetadata` |  no | | Plan metadata
+parameters | `[]Parameter` |  no | | Plan parameters
+
+**PlanMetadata**
+
+Field Name | Type | Required | Default | Description
+---|---|---|---|---
+displayName | string | no | "" | Name of plan for display purposes
+longDescription | string | no | "" | A detailed description of the plan
+cost | string | no | "" | Cost string used for display purposes only
+
+---
+
+3) Move any example of a `[]map[string]T` to `[]T`. The former introduced an entirely
+unecessary map layer. They started to get nested after moving params onto plans and
+things got very ugly because of it. Reintroduce `name` onto `T`s.
+
+**Old Format**
+
+```yaml
+parameters:
+  - mediawiki_db_schema:
+      default: mediawiki
+      type: string
+      title: Mediawiki DB Schema
+
+  - mediawiki_site_name:
+      default: MediaWiki
+      type: string
+      title: Mediawiki Site Name
+```
+
+**New Format**
+
+```yaml
+parameters:
+  - name: mediawiki_db_schema
+    default: mediawiki
+    type: string
+    title: Mediawiki DB Schema
+  - name: mediawiki_site_name
+    default: MediaWiki
+    type: string
+    title: Mediawiki Site Name
+```
+
+Should use this pattern for any additional changes moving forward.
+
+---
+
+4) Remove `required` section. `required` field should return to being an attribute on `Parameter`s.
+
+```yaml
+parameters:
+  - name: mediawiki_db_schema
+    required: true
+    default: mediawiki
+    type: string
+    title: Mediawiki DB Schema
+```
+
+## Notes
+
+* Use yaml anchors and inheritence to cut down on parameter duplication if plans share
+parameters, see full example.
+
+## Questions
+
+* Obviously this is a large, breaking change to **all** existing APBs and their `apb.yml` files.
+How do we want to make across the board so things get rolled out as smoothly as possible?
+* Is this a good opportunity to have the broker start respecting the version label found on
+apbs? I think it makes sense to have that in lock setup with broker release versions. Is
+broker master considered `0.10` since it's `release-0.9++`?
```

Comment:
The version in the ansible-service-broker rpm spec is currently 1.0.0 in master. We have not built an rpm from this spec yet so we can change it if we want to go do 0.10.0